jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/Configuration.java

```diff
@@ -0,0 +1,221 @@
+        public void setAutoFragment(boolean autoFragment)
+        {
+            this.autoFragment = autoFragment;
+        }
+
+        @Override
+        public long getMaxFrameSize()
+        {
+            return maxFrameSize == null ? WebSocketConstants.DEFAULT_MAX_FRAME_SIZE : maxFrameSize;
+        }
+
+        @Override
+        public void setMaxFrameSize(long maxFrameSize)
+        {
+            this.maxFrameSize = maxFrameSize;
+        }
+
+        @Override
+        public int getOutputBufferSize()
+        {
+            return outputBufferSize == null ? WebSocketConstants.DEFAULT_OUTPUT_BUFFER_SIZE : outputBufferSize;
+        }
+
+        @Override
+        public void setOutputBufferSize(int outputBufferSize)
+        {
+            this.outputBufferSize = outputBufferSize;
+        }
+
+        @Override
+        public int getInputBufferSize()
+        {
+            return inputBufferSize == null ? WebSocketConstants.DEFAULT_INPUT_BUFFER_SIZE : inputBufferSize;
+        }
+
+        @Override
+        public void setInputBufferSize(int inputBufferSize)
+        {
+            this.inputBufferSize = inputBufferSize;
+        }
+
+        @Override
+        public long getMaxBinaryMessageSize()
+        {
+            return maxBinaryMessageSize == null ? WebSocketConstants.DEFAULT_MAX_BINARY_MESSAGE_SIZE : maxBinaryMessageSize;
+        }
+
+        @Override
+        public void setMaxBinaryMessageSize(long maxBinaryMessageSize)
+        {
+            this.maxBinaryMessageSize = maxBinaryMessageSize;
+        }
+
+        @Override
+        public long getMaxTextMessageSize()
+        {
+            return maxTextMessageSize == null ? WebSocketConstants.DEFAULT_MAX_TEXT_MESSAGE_SIZE : maxTextMessageSize;
+        }
+
+        @Override
+        public void setMaxTextMessageSize(long maxTextMessageSize)
+        {
+            this.maxTextMessageSize = maxTextMessageSize;
+        }
+    }
+
+    class ConfigurationCustomizer extends ConfigurationHolder implements Customizer
+    {
+        @Override
+        public void customize(Configuration configurable)
+        {
+            if (idleTimeout != null)
+                configurable.setIdleTimeout(idleTimeout);
+            if (writeTimeout != null)
+                configurable.setWriteTimeout(writeTimeout);
+            if (autoFragment != null)
+                configurable.setAutoFragment(autoFragment);
+            if (maxFrameSize != null)
+                configurable.setMaxFrameSize(maxFrameSize);
+            if (inputBufferSize != null)
+                configurable.setInputBufferSize(inputBufferSize);
+            if (outputBufferSize != null)
+                configurable.setOutputBufferSize(outputBufferSize);
+            if (maxBinaryMessageSize != null)
+                configurable.setMaxBinaryMessageSize(maxBinaryMessageSize);
+            if (maxTextMessageSize != null)
+                configurable.setMaxTextMessageSize(maxTextMessageSize);
+        }
```

Comment:
I can assure you that you add a field and you will forget to add an `if` statement here.
Why can't we pass the configuration object around rather than customizing it?

Reply:
To my knowledge, there is no technology that allows a field to be added and to have affect everywhere that it is needed.  Other code always has to be implemented to use the new field. This is no different.